### PR TITLE
add qgs download link in layer detail

### LINKED ISSUE #157: Create downloadable QGIS project on the fly
Create downloadable QGIS project on the fly that references GeoNode layers including setting the mime type to pop up in QGIS after downloading. This is similar to the .qlr file download for maps, but instead of providing the user with a QGIS .qlr, we will instead generate a QGIS project file (.qgs) which they can open locally on their computer. QGIS Backend only.

Zip the file for download

Will use https://github.com/kartoza/otf-project

### PR DESCRIPTION
**Pull request by boney-bun**
push to be reviewed and discussed as there still a few works to be done

fix #157

### CODE REVIEW

File: geonode/qgis_server/views.py [hunk 1]
```diff
@@ -740,3 +740,55 @@ def set_thumbnail(request, layername):
     }
     return HttpResponse(
         json.dumps(retval), content_type="application/json")
+
+
+def download_qgs(request, layername):
```
Comment (ismailsunni) [inline geonode/qgis_server/views.py]: I saw the method is very similaar with download_zip. The only different is the .qgs file included or excluded. Perhaps you can just extend the method of download_zip and add a parameter to include qgs file.

**Comment by boney-bun** [comment]
the working code is illustrated below:
adddownloadqgs157
as shown in the GIF: when open qgs using QGIS, the app will complain the broken link at the attribute `datasource` in qgs file. 
the `datasource` points to the server path. it needs to be modified to relative path.
i could easily change this value (as in #155), but maybe the best solution is to address the path on the `otf-project` repo.
let me know your opinion @Gustry @gubuntu.

File: geonode/qgis_server/views.py (hunk 1 above)
Comment (boney-bun) [reply geonode/qgis_server/views.py]: yes indeed. i can add additional parameter to the GET request and filter them in the `url.py`. thanks for the comment @ismailsunni

**Comment by gubuntu** [comment]
@boney-bun the project should not be referencing any file path. Instead it should contain the layer as a WFS or WCS layer, referencing the GeoNode url, so it can be opened from anywhere.

**Comment by boney-bun** [comment]
i'll close this PR as no longer relevant.
we use #322 to address #157

**Pull request closed without merging**